packages/itwin/tree-widget/src/test/trees/VisibilityTreeEventHandler.test.ts

```diff
@@ -36,124 +49,124 @@ describe("VisibilityTreeEventHandler", () => {
     dispose: sinon.fake(),
   };
 
-  const testVisibilityStatus: VisibilityStatus = {
-    state: "visible",
-    isDisabled: false,
-  };
-
   beforeEach(() => {
-    getVisibilityStatus.returns(testVisibilityStatus);
-    modelMock.setup((x) => x.getNode(moq.It.isAny())).returns(() => createSimpleTreeModelNode());
-    modelSourceMock.setup((x) => x.onModelChanged).returns(() => new BeUiEvent<[TreeModel, TreeModelChanges]>());
-    modelSourceMock.setup((x) => x.getModel()).returns(() => modelMock.object);
-    nodeLoaderMock.setup((x) => x.dataProvider).returns(() => dataProviderMock.object);
-    nodeLoaderMock.setup((x) => x.modelSource).returns(() => modelSourceMock.object);
+    nodeLoaderStub.loadNode.returns(EMPTY);
   });
 
   afterEach(() => {
-    modelSourceMock.reset();
-    modelMock.reset();
-    nodeLoaderMock.reset();
-    dataProviderMock.reset();
-    selectionHandlerMock.reset();
-    getVisibilityStatus.reset();
+    nodeLoaderStub.loadNode.reset();
     changeVisibility.reset();
+    onVisibilityChange.clear();
+    getVisibilityStatus.resetHistory();
+    sinon.restore();
   });
 
+  function setupTreeModel(nodeIds: string[], item?: PresentationTreeNodeItem) {
+    const model = new MutableTreeModel();
+    model.setChildren(
+      undefined,
+      nodeIds.map((nodeId) => {
+        return { ...createSimpleTreeModelNode(nodeId), isLoading: false, item: item ?? createSimpleTreeModelNode("node-item") };
+      }),
+      0,
+    );
+    const modelSource = new TreeModelSource(model);
+    nodeLoaderStub.modelSource = modelSource;
+    modelSource.modifyModel = () => {};
```

Comment:
Why is this needed? If it's added in order to resolve error that requires to enable `immer` patches, you can initialize `UIComponents` before tests.

Reply:
You were right, left it only for onNodeDoubleClicked() test cases